packages/discount-code-exporter/test/main.spec.js

```diff
@@ -57,70 +57,146 @@ describe('DiscountCodeExport', () => {
     })
   })
 
-  describe('::run', () => {
-    test('should fetch discount codes and output csv to stream', done => {
-      codeExport.config.exportFormat = 'csv'
-      const sampleCode = {
-        code: 'discount-code',
-        name: { en: 'some-discount-name' },
-        cartDiscounts: [{ id: 'cart-discount-1' }, { id: 'cart-discount-2' }],
-      }
-      codeExport._fetchCodes = jest.fn().mockImplementation(csvStream => {
-        csvStream.write(sampleCode)
-        return Promise.resolve()
-      })
-      const outputStream = streamtest.v2.toText((error, result) => {
-        const expectedResult = stripIndent`
-          code,name.en,cartDiscounts,groups
-          discount-code,some-discount-name,cart-discount-1;cart-discount-2,
-          `
-        expect(result).toEqual(expectedResult)
-        done()
+  describe('::setupHeaders', () => {
+    describe('with passed headers', () => {
+      test('should return return passed headers', () => {
+        expect(DiscountCodeExport.setupHeaders(['foo', 'bar'], 'de')).toEqual([
+          'foo',
+          'bar',
+        ])
       })
-      codeExport.run(outputStream)
     })
 
-    test('should fetch codes and output json to stream by default', done => {
-      const sampleCode = {
-        code: 'discount-code',
-        name: { en: 'some-discount-name' },
-        cartDiscounts: [{ id: 'cart-discount-1' }, { id: 'cart-discount-2' }],
-      }
-      codeExport._fetchCodes = jest.fn().mockImplementation(jsonStream => {
-        jsonStream.write(sampleCode)
-        return Promise.resolve()
+    describe('without passed headers', () => {
+      describe('when empty array', () => {
+        test('should return return passed headers', () => {
+          expect(DiscountCodeExport.setupHeaders([], 'de')).toMatchSnapshot()
+        })
       })
-      const outputStream = streamtest.v2.toText((error, result) => {
-        const expectedResult = [sampleCode]
-        expect(JSON.parse(result)).toEqual(expectedResult)
-        done()
+
+      describe('when null', () => {
+        test('should return return passed headers', () => {
+          expect(DiscountCodeExport.setupHeaders(null, 'de')).toMatchSnapshot()
+        })
       })
-      codeExport.run(outputStream)
     })
+  })
+
+  describe('::run', () => {
+    describe('successful export', () => {
+      let sampleCode
+      beforeEach(() => {
+        sampleCode = {
+          code: 'discount-code',
+          name: { en: 'some-discount-name' },
+          cartDiscounts: [{ id: 'cart-discount-1' }, { id: 'cart-discount-2' }],
+        }
+        codeExport._fetchCodes = jest.fn().mockImplementation(csvStream => {
```

Comment:
Why `jest.fn().mockImplementation`? I know the docs state so but if we never `mockImplementation` later on wouldn't a `jest.fn((csvStream => {}))` be enough? Even then we could `mockImplementation` later?